### PR TITLE
LPS-148805 

### PR DESCRIPTION
**Pull request by leticia-maciel**
Avoid unpublished and saved forms to show on search results on Forms widget.

### CODE REVIEW

**Comment by leticia-maciel** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Review by rodrigopaulino — COMMENTED** [review]
Hey, @leticia-maciel.

Thanks for working on this. Could you please explain why did you create all these different find methods if you plan to only use one?

Looking forward to hearing from you.

**Comment by leticia-maciel** [comment]
Hey @rodrigopaulino ,
I had to create a new find method for every case with the new published variable.

**Comment by rodrigopaulino** [comment]
> Hey @rodrigopaulino , I had to create a new find method for every case with the new published variable.

Thanks for replying, @leticia-maciel. I think you can leave out the ones you don't use. Just create the ones you do use. Does it make sense?

**Comment by leticia-maciel** [comment]
ci:test:relevant

**Comment by leticia-maciel** [comment]
ci:test:relevant

**Comment by leticia-maciel** [comment]
> Hey, @leticia-maciel. Still I see that there's two new search methods in [DDMFormInstanceLocalService.java](https://github.com/liferay-forms/liferay-portal/pull/1707/files#diff-225a480361deb380504aeffad0a49d64fbf89a276cdd271e27f921ea695ddea6). Why did you have to do that? I can see that you're just using one of them: [configuration.jsp](https://github.com/liferay-forms/liferay-portal/pull/1707/files#diff-5923c6566561ce725b53fa08db023541947f164c0fca4cedcd28b0ea418694a4)

Hey @rodrigopaulino, you're right, I forgot to delete the modifications I did in DDMFormInstanceLocalServiceImpl.java. I've already correct it. 

Thanks :grin:

**Comment by leticia-maciel** [comment]
ci:test:sf

**Comment by leticia-maciel** [comment]
ci:test:relevant

**Comment by leticia-maciel** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/resources/META-INF/custom-sql/default.xml [hunk 1]
```diff
@@ -100,6 +117,34 @@
...
+						DDMFormInstance, DDMFormInstanceVersion
+					WHERE
+						(DDMFormInstance.formInstanceId = DDMFormInstanceVersion.formInstanceId) AND
+						(DDMFormInstance.groupId = ?) AND
+						(DDMFormInstance.companyId = ?) AND
+						(
+							(LOWER(DDMFormInstance.name) LIKE ? [$AND_OR_NULL_CHECK$]) [$AND_OR_CONNECTOR$]
+							(DDMFormInstance.description LIKE ? [$AND_OR_NULL_CHECK$])
+						) AND
+						(DDMFormInstance.settings_ LIKE [$PUBLISHED_FILTER_BEGIN$][$PUBLISHED$][$PUBLISHED_FILTER_END$])
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/resources/META-INF/custom-sql/default.xml:138]: Can we put all the hardcoded LIKE filter already in the query? We just leave the [$PUBLISHED$] there.

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Hey, @leticia-maciel. Good work here!
Just have one final note down below.
Thanks for your contribution!!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/resources/META-INF/custom-sql/default.xml (hunk 1 above)
Comment (leticia-maciel) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/resources/META-INF/custom-sql/default.xml]: Done!

**Review by rodrigopaulino — APPROVED** [review]
LGTM! 🎉

**Comment by rodrigopaulino** [comment]
ci:forward

**Pull request closed without merging**